### ISSUE TITLE
Fuzzing for phragmen 

### ISSUE THREAD

**Issue by kianenigma** [paritytech/substrate#4632]
There should be a fuzzer that does the following: 

1. (**Important -- needed ASAP**) Generate a random election edge vector, as such

```rust 
let mut assignments = vec![
		StakedAssignment {
			who: 1, // voter
			distribution: vec![(10, 10)] // (target, weight)
		},
		StakedAssignment {
			who: 2,
			distribution: vec![
				(10, 15),
				(20, 5),
			],
		},
     ...
]
```

pass a cloned version of it through the `reduce()` function, then assert that it is equal to the original value. This can be easily done by using `build_support_map` and comparing the results. This means that the reduction algorithm, as expected, will just remove some edges but has no effect on the final outcome. I have written a function in my other PR to help with this:  

https://github.com/paritytech/substrate/blob/ca68e24d1879026a4c1b8f2123bd5caaef613f4b/primitives/phragmen/src/mock.rs#L410-L438

Basically just run this over a sensibly large random input. 

There should also be an assertion that the encoded length of the result is always less than or equal to that of the original one. We cannot make this strict to just less than since the random generation might actually lead to some graphs that cannot be reduced. A very good bonus is to tell us the size of reduced on average by how many percent. 

2. Likewise, the main `elect` function can also be compared to the float implementation. This has less priority at the moment. There is also a helper for this: `run_and_compare` in `primitives/phragmen/mock.rs` which is doing exactly that. The It just need to be fed some sensible data. 

cc @shawntabrizi

**Comment by kianenigma** [comment]
PR need to made against https://github.com/paritytech/substrate/pull/4517

**Comment by expenses** [comment]
@kianenigma I'll definitely try and do this if I have time, but I can't make any guarantees, and I'd prefer not to be assigned to it. Thanks!

**Comment by kianenigma** [comment]
Already pushed to https://github.com/paritytech/substrate/pull/4517
